lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -246,5 +286,79 @@ def cleanup
       @logger.notify "Spent %.2f seconds cleaning up" % (Time.now - start)
     end
 
+    def add_disk(hostname, size)
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/api/v1/vm/' + hostname + '/disk/' + size.to_s)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Post.new(uri.request_uri)
+
+        request['X-AUTH-TOKEN'] = @credentials[:vmpooler_token]
+
+        response = http.request(request)
+      rescue RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        @logger.notify "Failed to connect to vmpooler for additional disk!"
```

Comment:
@eputnam I was curious to know why you chose to have much of this code rescue errors and use the logger to notify that disk provisioning failed. Since failed disk provisioning will certainly cause test failures with tests that rely on it, why not fail earlier?

In code that is similar in this class—the tagging code—beaker doesn't fail out, but that is because failed tagging doesn't affect actual testing, just our metrics around the testing itself.